app/models/host/managed.rb

```diff
@@ -653,6 +654,22 @@ def clone
     host
   end
 
+  def check_interfaces
+    primary_flag = false
+    provision_flag = false
+    self.interfaces.each do |interface|
+      if interface.primary
+        primary_flag = true
+      end
+      if interface.provision
+        provision_flag = true
+      end
+    end
+    if !provision_flag || !primary_flag
```

Comment:
This can be simplified to something like (pseudocode):

    interfaces.detect(:&primary) || interfaces.detect(:&provision)

I suggest to report the errors separately so user exactly knows which one of the two is missing.